src/main/java/org/craftercms/engine/servlet/filter/SiteAwareCORSFilter.java

```diff
@@ -0,0 +1,70 @@
+package org.craftercms.engine.servlet.filter;
+
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletResponseWrapper;
+
+import org.apache.commons.configuration2.HierarchicalConfiguration;
+import org.craftercms.commons.web.CORSFilter;
+import org.craftercms.engine.service.context.SiteContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+/**
+ * Filter to include CORS headers based on the current site configuration.
+ *
+ * @author Jose Ross
+ */
+public class SiteAwareCORSFilter extends OncePerRequestFilter {
+
+    private static final Logger logger = LoggerFactory.getLogger(SiteAwareCORSFilter.class);
+
+    public static final String CONFIG_KEY = "cors";
+    public static final String ENABLE_KEY = "[@enable]";
```

Comment:
Change the enabled flag to be an element instead of an attribute. This will let us later add YAML as a configuration file format and keep both YAML and XML without adding any code to support special selectors.